File: Introduction-to-Recursion/Power_Linear.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) {
        // write your code here
        Scanner scn = new Scanner(System.in);
        int n = scn.nextInt();
        int x = scn.nextInt();
        int ans = power(x, n);
        System.out.println(ans);
    }

    public static int power(int x, int n) {
        if (x == 0) {
            return 1;
        }
        int xM1power = power(x - 1, n);
        int npower = n * xM1power;
        return npower;
    }

}

File: 6. Introduction-to-Recursion/cpp/Power_Linear.cpp
#include<bits/stdc++.h>
using namespace std;
int power(int x,int n){
    if (x == 0) {
        return 1;
    }
    int xM1power = power(x - 1, n);
    int npower = n * xM1power;
    return npower;
}

int main(){
    int n,x;
    cin >> n;
    cin >> x;
    cout << power(x,n);

    return 0;
}